lib/pure/hashes.nim

```diff
@@ -114,7 +114,11 @@ proc hashWangYi1*(x: int64|uint64|Hash): Hash {.inline.} =
   const P1  = 0xe7037ed1a0b428db'u64
   const P58 = 0xeb44accab455d165'u64 xor 8'u64
   when nimvm:
-    cast[Hash](hiXorLo(hiXorLo(P0, uint64(x) xor P1), P58))
+    when defined(js): # NOTE: Nim int64<->Number => JS has limited 32-bit hash.
+      result = cast[Hash](hiXorLo(hiXorLo(P0, uint64(x) xor P1), P58)) and
```

Comment:
subexpression `cast[Hash](hiXorLo(hiXorLo(P0, uint64(x) xor P1), P58))` repeated 3 times; factor it using some auxiliary template

Reply:
Ok.  I kept the `cast[Hash]()` part open coded since it is short and there is one spot where it seems more clear to say `cast[Hash](a) and cast[Hash](b)` and have the type matching for the binary operator be obvious.